src/clients/FtpClient.cc

```diff
@@ -763,6 +763,8 @@ Ftp::Client::connectDataChannel()
     conn->remote.port(data.port);
     conn->tos = ctrl.conn->tos;
     conn->nfmark = ctrl.conn->nfmark;
+    // Using non-local addresses in TPROXY mode requires appropriate socket option.
+    conn->flags = ctrl.conn->flags & COMM_TRANSPARENT;
```

Comment:
getOutgoingAddress(), which works with somewhat related concepts, also sets COMM_DOBIND. Do we need that flag here as well?

Reply:
On our setup everything works without COMM_DOBIND flag.

Reply:
DOBIND is only necessary if the IP address might be the magic ANY_ADDR.  Which this FtpClient code is never permitting to happen (the ctrl->local should never be ANY_ADDR).

Reply:
@k0rv1n, It would still be good to use  "  |=  " to assign the COMM_TRANSPARENT flag without overwriting any existing defaults that might be there (now or in future).